### PR TITLE
Keep track of VPAs with no recommendation.

### PR DESCRIPTION
**Pull request by bskiba**

### CODE REVIEW

**Comment by bskiba** [comment]
@kgolab

File: vertical-pod-autoscaler/pkg/recommender/routines/recommender.go [hunk 1]
```diff
@@ -105,8 +107,19 @@ func (r *recommender) UpdateVPAs() {
...
 				metrics_recommender.ObserveRecommendationLatency(vpa.Created)
 			}
+			delete(r.clusterState.EmptyVPAs, vpa.ID)
 		} else {
 			vpa.Conditions.Set(vpa_types.RecommendationProvided, false, "", "")
+			firstObserved, ok := r.clusterState.EmptyVPAs[vpa.ID]
+			if !ok {
+				r.clusterState.EmptyVPAs[vpa.ID] = time.Now()
+			} else {
+				if firstObserved.Add(RecommendationMissingMaxDuration).Before(time.Now()) {
```
Comment (kgolab) [inline vertical-pod-autoscaler/pkg/recommender/routines/recommender.go]: I'd change the clusterState.EmptyVPAs meaning slightly so keep track the time from the last warning (and reset it inside this if) so we'd emit a warning for a problematic VPA twice an hour, not every minute. WDYT?

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/recommender/routines/recommender.go]: Done.

**Comment by bskiba** [comment]
I've reworked this change a little, added unit test (and rightly so as they caught some bugs...). PTAL

File: vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go [hunk 2]
```diff
@@ -386,3 +388,78 @@ func TestEmptySelector(t *testing.T) {
 	assert.Contains(t, vpa.aggregateContainerStates, cluster.aggregateStateKeyForContainerID(containerID1))
 	assert.Contains(t, vpa.aggregateContainerStates, cluster.aggregateStateKeyForContainerID(containerID2))
 }
+
+func TestRecordRecommendation(t *testing.T) {
+	testTimestamp := time.Unix(1555400, 0)
```
Comment (kgolab) [inline vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go]: Any reason why this particular timestamp?

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go]: Nah, removed. I didn't notice there is already a testTimestamp available

**Comment by kgolab** [comment]
Thanks, LGTM.

**Pull request merged**